metron-platform/metron-test-utilities/src/main/java/org/apache/metron/test/mock/MockHTable.java

```diff
@@ -437,15 +462,14 @@ public ResultScanner getScanner(byte[] family, byte[] qualifier)
     return getScanner(scan);
   }
 
-  List<Put> putLog = new ArrayList<>();
-
   public List<Put> getPutLog() {
-    return putLog;
+    return ImmutableList.copyOf(putLog);
```

Comment:
In changing to a `synchronizedList`, I don't think the defensive copy is thread-safe any longer.  I'm going to close this PR and test the whole thing holistically, since it is different enough from what I originally submitted.